### PR TITLE
Future fix for semaphore_406e

### PR DESCRIPTION
**Pull request by elad335**
Fix pain in the future.

### CODE REVIEW

**Comment by Nekotekina** [comment]
Better would be to initialize `be_t<u32> arg_be = arg;` and then compare arg_be with data loaded from memory in loop. If you want to optimize it so much, could just add raw() method to se_t class templates and compare raw values.

File: Utilities/BEType.h [hunk 1]
```diff
@@ -490,6 +490,11 @@ class se_t<T, true, Align>
 		return storage::from(m_data);
 	}
 
+	type raw() const
```
Comment (Nekotekina) [inline Utilities/BEType.h]: It should just return m_data. Preferably reference to, like in atomic_t. Casting it to the type is not meaningful.

File: rpcs3/Emu/RSX/rsx_methods.cpp [hunk 2]
```diff
@@ -64,17 +64,15 @@ namespace rsx
 			rsx->sync_point_request = true;
 			const u32 addr = get_address(method_registers.semaphore_offset_406e(), method_registers.semaphore_context_dma_406e());
 
-#ifdef IS_LE_MACHINE
-			arg = se_storage<u32>::swap(arg);
-			const auto& sema = vm::_ref<le_t<u32>>(addr);
-#else
+			// Preswap value
+			arg = static_cast<u32>(be_t<u32>(arg));
```
Comment (Nekotekina) [inline rpcs3/Emu/RSX/rsx_methods.cpp]: This expression doesn't actually swap bytes. I recommended to create a new (const) variable.

Comment (elad335) [reply rpcs3/Emu/RSX/rsx_methods.cpp]: Neither methods was working as expected, so I used raw() to make a one-direction swap.
I'm open to make this a method from se_storage like se_storage<>::get<bool is_be>.

File: rpcs3/Emu/RSX/RSXThread.cpp [hunk 3]
```diff
@@ -325,7 +325,7 @@ namespace rsx
 	{
 		//Endianness is swapped because common upload code expects input in BE
 		//TODO: Implement fast upload path for LE inputs and do away with this
-		element_push_buffer.push_back(se_storage<u32>::swap(index));
+		element_push_buffer.push_back(be_t<u32>{index}.raw());
```
Comment (kd-11) [inline rpcs3/Emu/RSX/RSXThread.cpp]: Need this swapped for now

File: rpcs3/Emu/RSX/rsx_methods.cpp [hunk 4]
```diff
@@ -244,8 +242,8 @@ namespace rsx
 			{
 			case rsx::vertex_base_type::ub:
 			case rsx::vertex_base_type::ub256:
-				// One-way byteswap
-				arg = se_storage<u32>::swap(arg);
+				// Get BE data
+				arg = be_t<u32>{arg}.raw();
```
Comment (kd-11) [inline rpcs3/Emu/RSX/rsx_methods.cpp]: Same here

File: rpcs3/Emu/RSX/RSXThread.cpp (hunk 3 above)
Comment (elad335) [reply rpcs3/Emu/RSX/RSXThread.cpp]: It is swapped, I'm using be_t for it to obey IS_BE_MACHINE macro.

Comment (kd-11) [reply rpcs3/Emu/RSX/RSXThread.cpp]: Doesn't .raw() just return index with the default endianness? Or maybe I'm misunderstanding its use?

Comment (elad335) [reply rpcs3/Emu/RSX/RSXThread.cpp]: The constructor *and* the converter byteswap, so by using raw() I'm making it only byteswap at the constructor, effectively.

**Comment by Snakegodeater** [comment]
I still get this error with God of War anscension !

**Comment by windyboy1704** [comment]
@snakegodeater It does not hurt anything, provide the log if you have problems with that

**Comment by DartPower** [comment]
Junks the log file with
0:01:04.785396 {PPU[0x1000018] Thread (DLC) [0x022c133c]} sys_fs: sys_fs_readdir(fd=49, dir=*0xd01c7c00, nread=*0xd01c7ad8)
1 minutes of play and logfile are 150+ mb, 10 minutes - ~4 gb.
image

**Pull request merged**